### PR TITLE
fix(curriculum): added new test and corrected existing solution for Remove Elements from a Linked List challenge

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

While reviewing PR #35825, I noticed the `remove` method (which is not used during the tests) was incorrect in that it would error out if an element passed to the function did not exist in the linked list.

This PR adds/correct the existing tests and provides a valid solution.

### CODE REVIEW

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/remove-elements-from-a-linked-list.english.md [hunk 1]
```diff
@@ -87,74 +89,70 @@ function LinkedList() {
...
+    var currentNode = head;
 
-    if (previous) {
-      previous.next = current.next;
+    while (currentNode.next !== null && currentNode.element !== element) {
+      previous = currentNode;
+      currentNode = currentNode.next;
+    }
+    
+    if (currentNode.next === null) {
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/remove-elements-from-a-linked-list.english.md]:
```suggestion
    if (currentNode.next === null && currentNode.element !== element) {
```
This lets you remove the last element in the list.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
I found a couple of minor issues - comments below.

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Thanks @RandellDawson LGTM now!

**Review by Manish-Giri — APPROVED** [review]
LGTM. Thank You for your contribution. 💯

**Pull request merged**